proposals/lip-0053.md

```diff
@@ -439,22 +440,27 @@ crossChainUpdateTransactionParams = {
 
 ```python
 def verify(ccu: Transaction) -> None:
+    certificateBytes = ccu.params.certificate
+    if len(certificateBytes) == 0 and ccu.params.inboxUpdate is empty: # An empty object has all properties set to their default values.
+        raise Exception("A cross-chain update must contain a non-empty certificate and/or a non-empty inbox update.")
+
     if not isLive(ccu.params.sendingChainID):
         raise Exception("The sending chain is not live.")
-
-    # Liveness condition is only checked on the mainchain for the first CCU with a non-empty inbox update.
-    if chainAccount(ccu.params.sendingChainID).status == CHAIN_STATUS_REGISTERED:
-        verifyLivenessConditionForRegisteredChains(ccu)
-    certificateBytes = ccu.params.certificate
+    
     if chainAccount(ccu.params.sendingChainID).status == CHAIN_STATUS_REGISTERED and len(certificateBytes) == 0:
         raise Exception("The first CCU must contain a non-empty certificate.")
```

Comment:
This was already wrong before this commit but could be fixed now: `ccu` must not be the first CCU even the condition `if chainAccount(ccu.params.sendingChainID).status == CHAIN_STATUS_REGISTERED` is fulfilled. So the text should be "CCUs from chains with status CHAIN_STATUS_REGISTERED must have non-empty certificates."

Reply:
fixed